src/components/decision-graph/graph/graph-aside.tsx

```diff
@@ -67,6 +70,41 @@ export const GraphAside: React.FC<GraphAsideProps> = ({ defaultOpenMenu = 'compo
     reader.readAsText(Array.from(fileList)?.[0], 'UTF-8');
   };
 
+  const uploadJDMExcel = (event: any) => {
+    const file = event?.target?.files[0];
+    const fileReader = new FileReader();
+
+    fileReader.readAsArrayBuffer(file);
+
+    fileReader.onload = async () => {
+      const buffer = fileReader.result as ArrayBuffer;
+
+      if (!buffer) return;
+
+      const nodesFromExcel = await readFromExcel(buffer);
+
+      const updatedNodes = decisionGraph.nodes.map((node) => {
+        let _node = node;
+        // updating existing nodes
+        nodesFromExcel.forEach((excelNode) => {
+          if (excelNode.id === node.id) _node = { ...node, content: excelNode.content };
+        });
+
+        return _node;
+      });
+
+      // filtering new nodes and setting them proper position
+      const newNodes = nodesFromExcel
+        .filter((node) => !updatedNodes.some((existingNode) => existingNode.id === node.id))
+        .map((newNode, index) => ({ ...newNode, position: { x: index * 250, y: 0 } }));
+
+      setDecisionGraph({
+        nodes: [...updatedNodes, ...newNodes],
+        edges: decisionGraph.edges,
+      });
+    };
```

Comment:
Might be good to wrap in try / catch. On success it should show message "Excel uploaded!" and on error "Failed to upload Excel" or similar.